catalog/transformer/catalog-transformer-preview/src/main/java/org/codice/ddf/transformer/preview/PreviewMetacardTransformer.java

```diff
@@ -34,15 +56,80 @@ public BinaryContent transform(Metacard metacard, Map<String, Serializable> argu
     }
 
     String preview = "No preview text available.";
+
     if (metacard.getAttribute(Extracted.EXTRACTED_TEXT) != null
         && metacard.getAttribute(Extracted.EXTRACTED_TEXT).getValue() != null) {
       preview =
           StringEscapeUtils.escapeHtml4(
                   metacard.getAttribute(Extracted.EXTRACTED_TEXT).getValue().toString())
               .replaceAll("[\n|\r]", "<br>");
-      preview = String.format("<head><meta charset=\"utf-8\"/>%s</head>", preview);
+    } else if (previewFromMetadata
+        && xPathExpressions != null
+        && StringUtils.isNotEmpty(metacard.getMetadata())) {
+      String text =
+          xPathExpressions
+              .stream()
+              .map(x -> evaluateXPath(metacard.getMetadata(), x))
+              .filter(Objects::nonNull)
+              .findFirst()
+              .orElse("");
+      if (StringUtils.isNotEmpty(text)) {
+        preview = StringEscapeUtils.escapeHtml4(text).replaceAll("[\n|\r]", "<br>");
+      }
     }
 
+    preview = String.format("<head><meta charset=\"utf-8\"/>%s</head>", preview);
+
     return new BinaryContentImpl(IOUtils.toInputStream(preview));
   }
+
+  private String evaluateXPath(String metadata, XPathExpression xPathExpression) {
+    StringReader metadataReader = new StringReader(metadata);
+    InputSource inputXml = new InputSource(metadataReader);
+
+    String text = null;
+    try {
+      Node result = (Node) xPathExpression.evaluate(inputXml, XPathConstants.NODE);
+      if (result != null && result.getTextContent() != null) {
+        text = result.getTextContent().trim();
+      }
+    } catch (XPathExpressionException e) {
+      LOGGER.error("Error evaluating xpath");
+    }
+
+    return text;
+  }
+
+  private List<XPathExpression> createXPathExpressions(List<String> previewElements) {
+    XPathFactory xPathFactory = XPathFactory.newInstance();
+    List<XPathExpression> xPathExpressions = new ArrayList<>();
+
+    String xPathString = "//*[name()='%s']";
```

Comment:
Change the variable name to let readers know what you are doing, or extract to method. (See below)